scalableminds/webknossos
# Keep zoom during rotation using the buttons of the 3D-viewport

This PR reworks the functionality of the rotation buttons of the 3D-Viewport. The zoom is now kept during rotation and the calculation is now shorted, commented and easier to understand.

### URL of deployed dev instance (used for testing):
- https://keepzoomduringtdbuttonrotation.webknossos.xyz

### Steps to test:
- View a dataset. The initial view of the 3d-viewport should show all three planes (independent of the z coordinate of the flycam). The whole width of the dataset should be displayed in the viewport.
- Now rotate the dataset in the viewport a bit and change the zoom and use the rotation buttons as you like. The rotation of the camera should always be as expected.
- One exception: The rotation of the buttons might look a bit off due to not using always the shortest path because we do not use quaternions for interpolation.

### Issues:
- fixes #4739

------
